src/main.rs

```diff
@@ -75,11 +95,36 @@ fn ci_env_set() -> bool {
     env::var("CI").is_ok()
 }
 
+fn user_repo_from_url(url: Url) -> Result<(String, String), String> {
```

Comment:
A tuple containing two strings. This allows for multiple return values and we extract both user and repo name at the same time